src/main/java/app/controller/soundEngine/SoundBlockingWall.java

```diff
@@ -33,7 +36,60 @@ public void draw(GraphicsContext gc)
     @Override
     public Vector intersection(SoundRay soundRay)
     {
-        return Intersection.findIntersection(a,b, soundRay.getStart(), soundRay.getEnd());
+        return Intersection.findIntersection(a,b,soundRay.getStart(), soundRay.getEnd());
+    }
+
+    @Override
+    public boolean intersects(SoundRay soundRay) {
+        return Intersection.hasIntersection(a,b,soundRay.getStart(), soundRay.getEnd());
+    }
+
+    @Override
+    public boolean onSegment(Vector point) {
+        // det(AB,AM) != 0 -> AB and AM are not parallel, so M cant be on AB
+        if(b.sub(a).cross(point.sub(a)) != 0){
+            return false;
+        }
+
+        if(b.getX() == a.getX()){
+            double t = (point.getY() - a.getY()) / (b.getY() - a.getY());
+
```

Comment:
This space is very frustrating, it makes it look like you're using a bracketless IF statement, but then you see there is a line under.  Two options:

A:
```java
if(b.getX() == a.getX())
{
    double t = (point.getY() - a.getY()) / (b.getY() - a.getY());
    return  t <= 1 && t >= 0; 
}
else
{
    double t = (point.getX() - a.getX()) / (b.getX() - a.getX());
    return  t <= 1 && t >= 0;
}
```

Or B (which I prefer):
```java
double t;
if(b.getX() == a.getX())
    t = (point.getY() - a.getY()) / (b.getY() - a.getY());
else
    t = (point.getX() - a.getX()) / (b.getX() - a.getX());
    
return  t <= 1 && t >= 0;

```